include/deal.II/base/mpi_large_count.h

```diff
@@ -0,0 +1,450 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2022 - 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+/**
+ *
+ * BigMPICompat - a tiny MPI 4.x compatibility library
+ *
+ * Released under MIT license at https://github.com/tjhei/BigMPICompat
+ */
+#ifndef BIG_MPI_COMPAT_H
+#define BIG_MPI_COMPAT_H
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_MPI
+#  include <mpi.h>
+// required for std::numeric_limits used below.
+#  include <limits>
+#  ifndef MPI_VERSION
+#    error "Your MPI implementation does not define MPI_VERSION!"
+#  endif
+
+#  if MPI_VERSION < 3
+#    error "BigMPICompat requires at least MPI 3.0"
```

Comment:
Aren't we requiring MPI 3.0 in deal.II? We had this discussion a couple of months ago; but I don't remember the outcome...

Reply:
Yes, we do. This is a sanity check that the MPI_VERSION exported by mpi.h is okay (because this is used below to switch between implementations)